packages/adapters/wagmi/src/client.ts

```diff
@@ -358,40 +356,49 @@ export class WagmiAdapter extends AdapterBlueprint {
     return formatUnits(params.value, params.decimals)
   }
 
-  public syncConnectors(options: AppKitOptions, appKit: AppKit) {
-    this.addWagmiConnectors(options, appKit)
+  private async addWagmiConnector(connector: Connector, options: AppKitOptions) {
+    let provider: W3mFrameProvider | UniversalProvider | undefined = undefined
 
-    const connectors = this.wagmiConfig.connectors.map(connector => ({
-      ...connector,
-      chain: this.namespace
-    }))
+    const shouldFetchProvider =
+      connector.id === ConstantsUtil.WALLET_CONNECT_CONNECTOR_ID ||
+      connector.id === ConstantsUtil.AUTH_CONNECTOR_ID
 
-    const uniqueIds = new Set()
-    const filteredConnectors = connectors.filter(item => {
-      const isDuplicate = uniqueIds.has(item.id)
-      uniqueIds.add(item.id)
+    if (shouldFetchProvider) {
+      provider = (await connector.getProvider().catch(() => undefined)) as
+        | W3mFrameProvider
+        | UniversalProvider
+        | undefined
+    }
 
-      return !isDuplicate
+    this.addConnector({
+      id: connector.id,
+      explorerId: PresetsUtil.ConnectorExplorerIds[connector.id],
+      imageUrl: options?.connectorImages?.[connector.id] ?? connector.icon,
+      name: PresetsUtil.ConnectorNamesMap[connector.id] ?? connector.name,
+      imageId: PresetsUtil.ConnectorImageIds[connector.id],
+      type: PresetsUtil.ConnectorTypesMap[connector.type] ?? 'EXTERNAL',
+      info:
+        connector.id === ConstantsUtil.INJECTED_CONNECTOR_ID ? undefined : { rdns: connector.id },
+      provider,
```

Comment:
what happens if provider is undefined?

Reply:
This is how we do it atm, we don't add `provider` in wagmi adapter [here](https://github.com/reown-com/appkit/blob/main/packages/adapters/wagmi/src/client.ts#L383-L393). This is because we're getting the connector id and find the connector in wagmi config instead, you can check it [here](https://github.com/reown-com/appkit/blob/main/packages/adapters/wagmi/src/client.ts#L441)